src/components/Offer/Offer.tsx

```diff
@@ -0,0 +1,75 @@
+import { useState } from 'react';
+import { useAppSelector } from 'redux/store';
+import { withTranslation } from 'react-i18next';
+import labeling from 'containers/labeling';
+import CouponInput from 'containers/CouponInput/CouponInput.container';
+import Payment from 'components/Payment';
+import Header from 'components/Header';
+import SectionHeader from 'components/SectionHeader';
+import Footer from 'components/Footer';
+import CheckoutPriceBox from 'components/CheckoutPriceBox';
+import FreeOffer from 'components/FreeOffer';
+import { selectOffer } from 'redux/offerSlice';
+import { selectOnlyOrder } from 'redux/orderSlice';
+import {
+  StyledOfferBody,
+  StyledOfferWrapper,
+  StyledOfferDetailsAndCoupon,
+  StyledOfferCouponWrapper,
+  OfferCardWrapperStyled
+} from './OfferStyled';
+import OfferCheckoutCard from '../OfferCheckoutCard';
+import { OfferProps } from './Offer.types';
+
+const Offer = ({ onSubmit, onPaymentComplete, t }: OfferProps) => {
+  const [coupon, setCoupon] = useState('');
+  const { trialAvailable } = useAppSelector(selectOffer).offer;
```

Comment:
shouldn't selectOffer return a offer object? They we would have simply `useAppSelector(selectOffer)`